File: Projects/UniverseExplorer/app/src/main/java/a45125/myapplication/ImagePlanet.java
package a45125.myapplication;

import androidx.appcompat.app.AppCompatActivity;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.content.res.TypedArray;
import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toolbar;

public class ImagePlanet extends AppCompatActivity {
    private TextView planetName;
    private ImageView img;
    private int index;
    private Bundle extras;
    private Toolbar tb;
    private boolean changeImg = false;
    private TypedArray arrayPlanets,planet;

    @SuppressLint("ResourceType")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_image_planet);

        planetName = findViewById(R.id.planetNameImage);
        img = findViewById(R.id.planetpicBig);

        extras = getIntent().getExtras();
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        arrayPlanets = getResources().obtainTypedArray(R.array.planets);
        index = extras.getInt("idx");
        int idPlanet = arrayPlanets.getResourceId(index, -1);
        planet = getResources().obtainTypedArray(idPlanet);
        planetName.setText(planet.getString(0));
        img.setImageResource(planet.getResourceId(4,0));
        setTitle("Planet " + planetName.getText().toString() + " images ");


        img.setOnClickListener(new View.OnClickListener() {
            @SuppressLint("ResourceType")
            @Override
            public void onClick(View v) {
                if(!changeImg){
                    changeImg=true;
                    img.setImageResource(planet.getResourceId(5,-1));
                }else{
                    changeImg=false;
                    img.setImageResource(planet.getResourceId(4,-1));
                }
            }
        });
    }


    public void onBackPressed() {
        Intent intent = new Intent();
        setResult(RESULT_CANCELED, intent);
        finish();
    }

    public boolean onOptionsItemSelected(final MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                this.finish();
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }

}

File: Projects/UniverseExplorer/app/src/main/java/a45125/dam/universeexplorer/ImagePlanet.kt
package a45125.dam.universeexplorer

import android.annotation.SuppressLint
import android.content.Intent
import android.content.res.TypedArray
import android.os.Bundle
import android.view.MenuItem
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toolbar
import androidx.appcompat.app.AppCompatActivity

class ImagePlanet : AppCompatActivity() {
    lateinit var planetName: TextView
    lateinit var img: ImageView
    private var index = 0
    private var extras: Bundle? = null
    private val tb: Toolbar? = null
    private var changeImg = false
    private var arrayPlanets: TypedArray? = null
    private var planet: TypedArray? = null
    @SuppressLint("ResourceType")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_image_planet)
        planetName = findViewById(R.id.planetNameImage)
        img = findViewById(R.id.planetpicBig)
        extras = intent.extras
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        arrayPlanets = resources.obtainTypedArray(R.array.planets)
        index = extras!!.getInt("idx")
        val idPlanet = arrayPlanets!!.getResourceId(index, -1)
        planet = resources.obtainTypedArray(idPlanet)
        planetName.setText(planet!!.getString(0))
        img.setImageResource(planet!!.getResourceId(4, 0))
        title = "Planet " + planetName.getText().toString() + " images "
        img.setOnClickListener(View.OnClickListener {
            if (!changeImg) {
                changeImg = true
                img.setImageResource(planet!!.getResourceId(5, -1))
            } else {
                changeImg = false
                img.setImageResource(planet!!.getResourceId(4, -1))
            }
        })
    }

    override fun onBackPressed() {
        val intent = Intent()
        setResult(RESULT_CANCELED, intent)
        finish()
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            android.R.id.home -> {
                finish()
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }
}